### PR TITLE
implemented maps

### PR DESCRIPTION
**Pull request by tinawen02**
In this PR, I began implementing my Google Maps API. I created a new activity GoogleMapsActivity and wrote the code behind finding a user’s longitude and latitude. I also worked on displaying restaurants nearby to the user’s location, and set different colored pins on restaurants depending on the status of their hours (pink for open, orange for close, yellow for unsure). In addition to this, I also played around with the UI of my app. I fixed my splash screen to be cleaner, and brought in post-it tabs for each bill to be displayed on in the main feed. I had a few bugs with implementing the map - particularly for when the hours of operation could not be found for a restaurant. In this case, I ended up marking the pin as yellow. In following diffs, I want to implement a search feature for the user to search up food they want to eat.

I tested my changes by comparing the map my app displays with the map displayed in Google Maps and ensuring that both displayed the same restaurants. In addition to this, I checked the colors of the pins with the hours of operations the restaurants had displayed on Google to ensure everything was accurate.

https://user-images.githubusercontent.com/84735390/176503667-52b61893-f794-41d8-ad63-63d67f76b58b.mov

### CODE REVIEW

File: app/src/main/java/com/example/iou/map/activities/GoogleMapsActivity.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+
+        // Add a marker in Sydney and move the camera
+        /*
+        LatLng seattle = new LatLng(47.602, -122.3321);
+        mMap.addMarker(new MarkerOptions().position(seattle).title("Marker in Seattle"));
+        mMap.moveCamera(CameraUpdateFactory.newLatLng(seattle));
+
+         */
```
Comment (xiongamy) [inline app/src/main/java/com/example/iou/map/activities/GoogleMapsActivity.java:185]: commented out code

File: app/src/main/java/com/example/iou/map/activities/GoogleMapsActivity.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    private final static String KEY_LOCATION = "location";
+    private double longitude;
+    private double latitude;
+    private double restaurantLatitude;
+    private double restaurantLongitude;
+    private List<RestaurantItem> nearbyRestaurants;
+    Location mCurrentLocation;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
Comment (xiongamy) [inline app/src/main/java/com/example/iou/map/activities/GoogleMapsActivity.java:53]: This function is very long & the code inside gets pretty deeply nested. Would recommend splitting out some of the code in the listeners into separate functions!

File: app/src/main/java/com/example/iou/map/activities/GoogleMapsActivity.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            @Override
+            public void onLocationChanged(@NonNull Location location) {
+
+                // Retrieve the longitude and latitude of the user
+                longitude = location.getLongitude();
+                latitude = location.getLatitude();
+
+                // Add a pin at user's current location
+                LatLng userCoordinates = new LatLng(latitude, longitude);
+                mMap.addMarker(new MarkerOptions().position(userCoordinates).title("Your Current Coordinates"));
```
Comment (xiongamy) [inline app/src/main/java/com/example/iou/map/activities/GoogleMapsActivity.java:90]: You may need to add a check somewhere before this to see if `mMap` has been set yet

File: app/src/main/java/com/example/iou/map/models/RestaurantItem.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        catch (JSONException e) {
+            isOpen = null;
+        }
+        try {
+            this.rating = jsonObject.getDouble("rating");
+        }
+        catch (JSONException e) {
+            this.rating = -1;
+            e.printStackTrace();
+        }
```
Comment (xiongamy) [inline app/src/main/java/com/example/iou/map/models/RestaurantItem.java:36]: You can use `jsonObject.has()` to check if a field exists on the object, rather than having to wrap this code in try-catch blocks! There's also some optional versions of the get functions, e.g. `optDouble()`, that will returned the value if it exists, and a specified fallback value otherwise.

Android docs on JSONObject, for more info on the functions: https://developer.android.com/reference/org/json/JSONObject

File: app/src/main/java/com/example/iou/ui/bill/BillFragment.java [hunk 5]
```diff
@@ -53,6 +53,18 @@ public void onClick(View v) {
...
+        btnSplitEven.setOnHoverListener(new View.OnHoverListener() {
+            @Override
+            public boolean onHover(View v, MotionEvent event) {
+                Toast.makeText(getContext(), "Select this option if you want to split items between people!", Toast.LENGTH_SHORT).show();
+                return false;
+            }
+        });
+        
+         */
+
```
Comment (xiongamy) [inline app/src/main/java/com/example/iou/ui/bill/BillFragment.java:67]: commented out code

File: app/src/main/res/layout/activity_even_information.xml [hunk 6]
```diff
@@ -8,39 +8,43 @@
 
     <EditText
         android:id="@+id/etLocationEven"
-        android:layout_width="320dp"
```
Comment (xiongamy) [inline app/src/main/res/layout/activity_even_information.xml:11]: Happy to see you're playing around with the UI and customizing the app to your own liking! I think the post-it notes are very cute :)
These changes should be in a different commit & PR, though, since these changes are very distinct from the map changes!

**Review by xiongamy — COMMENTED** [review]
I like the different colored pins on the map! Very nice!

A couple of product questions:
1. I see in your video that right now to actually see the map, you have a press a "Click to Go to Map" button. Are you planning on having the map show up immediately on going to the Map tab later? (Were there some technical limitations that required the in-between button?)
2. Might be good to add a small legend in the corner that explains the pin colors! (Not necessary though, just a suggestion for a later addition!)

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -1,10 +1,26 @@
...
+    package="com.example.iou"
+    android:versionCode="1"
+    android:versionName="1.0">
+
+    <permission
+        android:name="android.permission.ACCESS_FINE_LOCATION"
+        android:protectionLevel="dangerous" />
+    <permission
+        android:name="android.permission.ACCESS_COARSE_LOCATION"
+        android:protectionLevel="dangerous" />
```
[the remark this answers is not in the document]
Comment (tinawen02) [reply app/src/main/AndroidManifest.xml:13]: I'm not too sure -I was just following documentation in the codepath guides!

[the remark this answers is not in the document]
Comment (tinawen02) [reply app/src/main/AndroidManifest.xml:13]: I'll look into it some more!

File: app/src/main/java/com/example/iou/bill/activities/SplitSettlementCalculatorActivity.java [hunk 8]
```diff
@@ -0,0 +1,4 @@
+package com.example.iou.bill.activities;
+
+public class SplitSettlementCalculatorActivity {
```
[the remark this answers is not in the document]
Comment (tinawen02) [reply app/src/main/java/com/example/iou/bill/activities/SplitSettlementCalculatorActivity.java]: I used it to move my logic to be separate from UI - it'll be shown in my next PR!
